server.js

```diff
@@ -1,30 +1,142 @@
+
+const port = process.env.PORT || 8080;
+const app = express();
+
+const thoughtSchema = new mongoose.Schema({
+  message: {
+    type: String,
+    required: [true, "Message is required"],
+    unique: true,
+    validate: {
+      validator: (value) => {
+        return /^[^0-9]+$/.test(value);
+      },
+      message: "Numbers are not allowed",
+    },
+    minlength: 5,
+    maxlength: 140,
+  },
+  hearts: {
+    type: Number,
+    default: 0,
+  },
+  createdAt: {
+    type: Date,
+    default: Date.now,
+  },
+});
+
+const Thought = mongoose.model("Thought", thoughtSchema);
+
+app.use(cors());
+app.use(express.json());
+
+app.get("/", (req, res) => {
+  res.send("MY HAPPY THOUGHTS");
+});
+
+app.get("/thoughts", async (req, res) => {
+  const allThougths = await Thought.find().sort({ createdAt: -1 }).limit(20);
+  res.json(allThougths);
+});
+
+app.post("/thoughts", async (req, res) => {
+  try {
+    const { message } = req.body;
+    const newThought = await new Thought({ message }).save();
+    res.json(newThought);
+  } catch (error) {
+    if (error.code === 11000) {
+      res
+        .status(400)
+        .json({ error: "Duplicated value", fields: error.keyValue });
+    }
+    res.status(400).json(error);
+  }
+});
+
+app.post("/thoughts/:id/likes", async (req, res) => {
+  const { id } = req.params;
+
+  try {
+    const updatedThought = await Thought.findByIdAndUpdate(
+      id,
+      {
+        $inc: { hearts: 1 },
+      },
+      { new: true }
+    );
+    if (updatedThought) {
+      res.json(updatedThought);
+    } else {
+      res.status(404).json({ message: "Not found" });
+    }
+  } catch (error) {
+    res.status(400).json({ message: "Invalid request", error });
+  }
+});
+
+app.delete("/thoughts/:id", async (req, res) => {
+  const { id } = req.params;
+
+  try {
+    const deletedThoughts = await Thought.findOneAndDelete({ _id: id });
+    if (deletedThoughts) {
+      res.json(deletedThoughts);
+    } else {
+      res.status(404).json({ message: "Not found" });
+    }
+  } catch (error) {
+    res.status(400).json({ message: "Invalid request", error });
+  }
+});
+
+app.patch("/thoughts/:id", async (req, res) => {
+  const { id } = req.params;
+
+  try {
+    const updatedThought = await Thought.findByIdAndUpdate(id, req.body, {
+      new: true,
+    });
+    if (updatedThought) {
+      res.json(updatedThought);
+    } else {
+      res.status(404).json({ message: "Not found" });
+    }
+  } catch (error) {
+    res.status(400).json({ message: "Invalid request", error });
+  }
+});
+
+app.put("/thoughts/:id", async (req, res) => {
+  const { id } = req.params;
+
+  try {
+    const updatedThought = await Thought.findOneAndReplace(id, eq.body, {
+      new: true,
+    });
+    if (updatedThought) {
+      res.json(updatedThought);
+    } else {
+      res.status(404).json({ message: "Not found" });
+    }
+  } catch (error) {
+    res.status(400).json({ message: "Invalid request", error });
+  }
+});
+
```

Comment:
Remember to restrict access to these endpoints because not all users should have access to these functionalities 🤓